--- Math-Pari-2.010603/utils/Math/PariBuild.pm.orig	2006-01-18 17:37:03.000000000 -0500
+++ Math-Pari-2.010603/utils/Math/PariBuild.pm	2006-01-18 17:37:38.000000000 -0500
@@ -712,7 +712,9 @@ sub find_machine_architecture () {
   } elsif ($os eq 'linux') {
     chomp($machine = `uname -m`);
     $machine = 'sparcv9' if $machine eq 'sparc64';
-    if (-e '/proc/cpuinfo') {
+	$machine = 'hppa' if $machine eq 'parisc';
+	$machine = 'hppa' if $machine eq 'parisc64';
+	if (-e '/proc/cpuinfo' && -R '/proc/cpuinfo') {
       open IN, '/proc/cpuinfo' or die "open /proc/cpuinfo: $!";
       local $/ = undef;		# Needed?
       my $info = <IN>;
